packages/ecs/src/UUIDComponent.test.tsx

```diff
@@ -0,0 +1,249 @@
+      assert.throws(() => {
+        setComponent(entity1, UUIDComponent)
+      }, Error)
+    })
+
+    it('should set/get the data of the component.', () => {
+      // Case1: set/get
+      setComponent(entity1, UUIDComponent, TestUUID)
+      const component1 = getComponent(entity1, UUIDComponent)
+      assert.ok(component1, 'The UUIDComponent did not get set correctly')
+      assert.equal(component1, TestUUID, 'The UUID value did not get set correctly')
+    })
+
+    it("shouldn't change the data when set multiple times with the same data", () => {
+      setComponent(entity1, UUIDComponent, TestUUID)
+      const component1 = getComponent(entity1, UUIDComponent)
+      setComponent(entity1, UUIDComponent, TestUUID)
+      const component2 = getComponent(entity1, UUIDComponent)
+      assert.equal(component1, component2)
+    })
+
+    it('Should throw an error when the UUID is already in use for another entity', () => {
+      setComponent(entity1, UUIDComponent, TestUUID)
+      assert.throws(() => {
+        setComponent(entity2, UUIDComponent, TestUUID)
+      }, Error)
+    })
+
+    it('should remove the old uuid from the entity', () => {
+      setComponent(entity1, UUIDComponent, TestUUID)
+      setComponent(entity1, UUIDComponent, TestUUID2)
+      assert.notEqual(getComponent(entity1, UUIDComponent), TestUUID)
+    })
+
+    it('should set a new uuid, and return its value when called with getOptionalComponent', () => {
+      setComponent(entity1, UUIDComponent, TestUUID)
+      assert.notEqual(getOptionalComponent(entity1, UUIDComponent), undefined)
+    })
+  })
+
+  describe('toJson', () => {
+    it('should return correctly serialized data', () => {
+      setComponent(entity1, UUIDComponent, TestUUID)
+      const json = serializeComponent(entity1, UUIDComponent)
+      assert.equal(json, TestUUID as string)
+    })
+  })
+
+  describe('onRemove', () => {
+    it('should remove the component from the entity', () => {
+      setComponent(entity1, UUIDComponent, TestUUID)
+      removeComponent(entity1, UUIDComponent)
+      assert.equal(UndefinedEntity, UUIDComponent.entitiesByUUIDState[TestUUID].value)
+      assert.equal(false, hasComponent(entity1, UUIDComponent))
+      assert.equal(getOptionalComponent(entity1, UUIDComponent), undefined)
+    })
+
+    it('should do nothing if the entity does not have the component', () => {
+      removeComponent(entity1, UUIDComponent)
+      assert.equal(UndefinedEntity, UUIDComponent.entitiesByUUIDState[TestUUID].value)
+      assert.equal(getOptionalComponent(entity1, UUIDComponent), undefined)
+    })
+  })
+
+  describe('getEntityByUUID', () => {
+    it('should return the correct entity', () => {
+      setComponent(entity1, UUIDComponent, TestUUID)
+      const testEntity = UUIDComponent.getEntityByUUID(TestUUID)
+      assert.equal(testEntity, UUIDComponent.entitiesByUUIDState[TestUUID].value)
```

Comment:
you can just pass entity1 into this condition as testEntity should be the same type and value as entity1.

Reply:
The idea of that `entitiesByUUIDState` check is to confirm that the underlying data is the same as the expected value, independent of what the function is doing internally
Added the comment as an extra check with c37140192ebfb47c3732e8c091ba64c751bd3638